x-pack/test/apm_api_integration/tests/transactions/transactions_groups_statistics.ts

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import url from 'url';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import archives from '../../common/fixtures/es_archiver/archives_metadata';
+import { registry } from '../../common/registry';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+
+  const archiveName = 'apm_8.0.0';
+  const { start, end } = archives[archiveName];
+  const transactionNames = ['DispatcherServlet#doGet', 'APIRestController#customers'];
+
+  registry.when(
+    'Transaction groups agg results when data is not loaded',
+    { config: 'basic', archives: [] },
+    () => {
+      it('handles the empty state', async () => {
+        const response = await supertest.get(
+          url.format({
+            pathname: `/api/apm/services/opbeans-java/transactions/groups/statistics`,
+            query: {
+              start,
+              end,
+              uiFilters: '{}',
+              numBuckets: 20,
+              latencyAggregationType: 'avg',
+              transactionType: 'request',
+              transactionNames: JSON.stringify(transactionNames),
+            },
+          })
+        );
+
+        expect(response.status).to.be(200);
+        expect(response.body).to.empty();
+      });
+    }
+  );
+
+  registry.when(
+    'Transaction groups agg results when data is loaded',
+    { config: 'basic', archives: [archiveName] },
+    () => {
+      it('returns the correct data', async () => {
+        const response = await supertest.get(
+          url.format({
+            pathname: `/api/apm/services/opbeans-java/transactions/groups/statistics`,
+            query: {
+              start,
+              end,
+              uiFilters: '{}',
+              numBuckets: 20,
+              transactionType: 'request',
+              latencyAggregationType: 'avg',
+              transactionNames: JSON.stringify(transactionNames),
+            },
+          })
+        );
+
+        expect(response.status).to.be(200);
+
+        expect(Object.keys(response.body).length).to.be(transactionNames.length);
+        expectSnapshot(Object.keys(response.body)).toMatchInline(`
+                  Array [
+                    "DispatcherServlet#doGet",
+                    "APIRestController#customers",
+                  ]
+              `);
+
+        expect(Object.values(response.body).map((group: any) => group.impact).length).to.be(2);
```

Comment:
Why 2? What's different here from just checking Object.keys(response.body).length?

Reply:
I removed this test, I'm now checking the impact of the first item.